### PR TITLE
com.github.alexkdeveloper.radio

### PR DESCRIPTION
**Pull request by alexkdeveloper**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.github.alexkdeveloper.radio.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "runtime-version" : "40",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "radio",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--filesystem=home"
```
Comment (hfiguiere) [inline com.github.alexkdeveloper.radio.json:13]: why does it need home access?

**Comment by hfiguiere** [comment]
bot, build com.github.alexkdeveloper.radio

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
appstream file doesn't validate.

File: com.github.alexkdeveloper.radio.json (hunk 1 above)
Comment (alexkdeveloper) [reply com.github.alexkdeveloper.radio.json:13]: The program saves stations in the file system.

Comment (hfiguiere) [reply com.github.alexkdeveloper.radio.json:13]: where? 

if it uses standard location for settings, then this is not needed.

Comment (alexkdeveloper) [reply com.github.alexkdeveloper.radio.json:13]: The application creates a hidden folder in the user's home directory. That's how it does it:

```
directory_path = Environment.get_home_dir()+"/.stations_for_radio_app";
   GLib.File file = GLib.File.new_for_path(directory_path);
   if(!file.query_exists()){
     try{
        file.make_directory();
     }catch(Error e){
        stderr.printf ("Error: %s\n", e.message);
     }
     create_default_stations();
   }
   show_stations();
```

Comment (alexkdeveloper) [reply com.github.alexkdeveloper.radio.json:13]: I changed appdata.xml.in. Added the <name>tag. I changed the commit in the json file.

Comment (hfiguiere) [reply com.github.alexkdeveloper.radio.json:13]: The application should conform to the xdg spec instead. In the mean time you can just use `--persist=.stations_for_radio_app` instead for granting home access.

Comment (alexkdeveloper) [reply com.github.alexkdeveloper.radio.json:13]: I tried it with persist=. stations_for_radio_app. No stations are created at startup. At the first start, the program should show a list of 10 stations. The list is empty. But you can create new stations and delete them manually. Is there any way to explicitly specify the folder with stations? For example, --filesystem=/. stations_for_radio_app or something similar?

Comment (alexkdeveloper) [reply com.github.alexkdeveloper.radio.json:13]: Maybe you can leave the permissions --filesystem=home ?

Comment (alexkdeveloper) [reply com.github.alexkdeveloper.radio.json:13]: My application works with this permission:
 --filesystem=home/. stations_for_radio_app 
Can I use it?

**Comment by hfiguiere** [comment]
bot, build com.github.alexkdeveloper.radio

**Comment by alexkdeveloper** [comment]
Hi! I applied Environment.get_user_data_dir (). Removed the filesystem=home permission.

**Comment by alexkdeveloper** [comment]
Now the code is like this:

`Environment.get_user_data_dir()+"/.stations_for_radio_app";`

The filesystem=home permission is no longer needed.

**Comment by nedrichards** [comment]
bot, build com.github.alexkdeveloper.radio

**Comment by nedrichards** [comment]
I still see --filesystem=home in the manifest?

**Pull request closed without merging**